sway-core/src/decl_engine/item_decl_id.rs

```diff
@@ -7,61 +7,61 @@ use crate::{
 };
 
 #[derive(Debug, Eq, PartialEq, Hash, Clone)]
-pub enum FunctionalDeclId {
+pub enum ItemDeclId {
```

Comment:
What about `AssociatedItemDeclId`? I know it's longer but it's not used that often and that's the [technical term](https://doc.rust-lang.org/reference/items/associated-items.html) items in `impl <>` and `impl <> for <>` blocks.